.github/workflows/release.yml

```diff
@@ -0,0 +1,37 @@
+name: Release
+
+on:
+  push:
+    tags: [v*]
+
+jobs:
+  sdist:
+    runs-on: ubuntu-latest
+    environment: release
+
+    permissions:
+      contents: write # Used to authenticate github release publish
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.19'
+
+    - name: Generate sdist
+      run: |
+        VERSION=${GITHUB_REF_NAME#v}
+        go mod vendor
```

Comment:
Let's not include vendored dependencies

Reply:
I disagree here. We do want to add dependencies as vendored, since it is part of what downstream (like Gentoo) need to perform offline source build. This was my main goal with this PR, an automated process to add vendored dependencies to the release tarball.

Reply:
Ok, I see.
I like the idea of including the dependencies as it gives a full snapshot of the release at tjhe time it was cut.
On the other hand "nobody else is doing this" and I think it's cause nobody else has the "offline build" requirement that Gentoo has.

Let's include the vendored deps for now and if it causes problems we can revisit.